src/UI/Component/Signal.php

```diff
@@ -12,4 +12,18 @@
  */
 interface Signal {
 
+	/**
+	 * Get the ID of this signal
+	 *
+	 * @return string
+	 */
+	public function getId();
```

Comment:
+1 for abetter solution at some point. +1 for currently, this is the best we have. and we life with it for now.